pyodide-build/pyodide_build/buildall.py

```diff
@@ -719,6 +723,15 @@ def copy_packages_to_dist_dir(
             output_dir / dist_artifact_path.name, compression_level=compression_level
         )
 
+        if metadata_files and dist_artifact_path.suffix == ".whl":
+            with ZipFile(dist_artifact_path, mode="r") as wheel:
+                name, ver, _ = dist_artifact_path.name.split("-", 2)
+                metadata_path = str(Path(f"{name}-{ver}.dist-info") / "METADATA")
```

Comment:
FYI, your proposed approach could be implemented as follows:
```python
metadata_infos = [
    info for info in wheel.infolist() if Path(info).name == "METADATA" and len(Path(info).parts) == 2
]

if len(metadata_infos) > 1:
    raise Exception(
        f"Unexpected number of METADATA files when building {pkg.name}: {len(metadata_infos)}"
    )
elif len(metadata_infos) == 1:
    metadata_info = metadata_infos[0]
    metadata_path = metadata_info.filename
    metadata_info.filename = f"{dist_artifact_path.name}.metadata"
    wheel.extract(metadata_path, output_dir)
```

Reply:
I think it is okay to raise an error and quit if some packages does not follow the PEP. I believe `pip install` will also fail on those cases.

Reply:
You can probably adapt how pip finds a .dist_info dir: https://github.com/pypa/pip/blob/ea727e4d6ab598f34f97c50a22350febc1214a97/src/pip/_internal/utils/wheel.py#L38

Reply:
> Probably it would be useful to check how pip finds a .dist_info dir: https://github.com/pypa/pip/blob/ea727e4d6ab598f34f97c50a22350febc1214a97/src/pip/_internal/utils/wheel.py#L38

I'm not sure I'm happy with my latest changes, as they increase the complexity a bit. However, we would now use the same functionality as pip.